From 9166c5d053fd0d798c130137d56ed54163f5906b Mon Sep 17 00:00:00 2001
From: Seth Jennings <sjenning@redhat.com>
Date: Mon, 18 Aug 2014 15:42:47 -0400
Subject: [edac] sb_edac: remove bogus assumption on mc ordering

Message-id: <1408376571-8244-9-git-send-email-sjenning@redhat.com>
Patchwork-id: 87870
O-Subject: [RHEL7.1 PATCH 08/12] sb_edac: remove bogus assumption on mc ordering
Bugzilla: 1127019
RH-Acked-by: Aristeu Rozanski <aris@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Aristeu Rozanski <aris@redhat.com>

Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7802951
Bugzilla: https://brewweb.devel.redhat.com/taskinfo?taskID=7802951
Upstream: adc61bcd91aa8364efaa496e7a4699950c362c21

When a MC is handled, the correct sbridge_dev is searched based on the node,
checking again later with the assumption the first memory controller found is
the first socket's memory controller is a bogus assumption. Get rid of it.

Cc: Tony Luck <tony.luck@intel.com>
Signed-off-by: Aristeu Rozanski <aris@redhat.com>
Signed-off-by: Mauro Carvalho Chehab <m.chehab@samsung.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/sb_edac.c b/drivers/edac/sb_edac.c
index ab12fb9..a5ef406 100644
--- a/drivers/edac/sb_edac.c
+++ b/drivers/edac/sb_edac.c
@@ -1807,10 +1807,6 @@ static int sbridge_mce_check_error(struct notifier_block *nb, unsigned long val,
      "%u APIC %x\n", mce->cpuvendor, mce->cpuid,
      mce->time, mce->socketid, mce->apicid);
 
- /* Only handle if it is the right mc controller */
- if (cpu_data(mce->cpu).phys_proc_id != pvt->sbridge_dev->mc)
-  return NOTIFY_DONE;
-
  smp_rmb();
  if ((pvt->mce_out + 1) % MCE_LOG_LEN == pvt->mce_in) {
   smp_wmb();
-- 
1.7.1